test/helpers/cons.go

```diff
@@ -240,6 +240,14 @@ const (
 	uninitializedRegen  = "Uninitialized regeneration level"                         // from https://github.com/cilium/cilium/pull/10949
 	unstableStat        = "BUG: stat() has unstable behavior"                        // from https://github.com/cilium/cilium/pull/11028
 	removeTransientRule = "Unable to process chain CILIUM_TRANSIENT_FORWARD with ip" // from https://github.com/cilium/cilium/issues/11276
+	// ...and their exceptions.
+	lrpExists            = "local-redirect service exists for frontend"                  // cf. https://github.com/cilium/cilium/issues/16400
+	opCannotBeFulfilled  = "Operation cannot be fulfilled on leases.coordination.k8s.io" // cf. https://github.com/cilium/cilium/issues/16402
+	lockDeletedEp        = "lock failed: endpoint is in the process of being removed"    // cf. https://github.com/cilium/cilium/issues/16422
+	listenAndServeFailed = "ListenAndServe failed for service health server"             // cf. https://github.com/cilium/cilium/pull/16477
+	globalDataSupport    = "kernel doesn't support global data"                          // cf. https://github.com/cilium/cilium/issues/16418
+	removingInexistantID = "removing identity not added to the identity manager!"        // cf. https://github.com/cilium/cilium/issues/16419
+	failedToListCRDs     = "the server could not find the requested resource"            // cf. https://github.com/cilium/cilium/issues/16425
```

Comment:
If I understand correctly, we will basically ignore these messages but they could still occur in real clusters and may represent real logical inconsistencies (possibly even bugs) in the agent code. What do you think is the best way to approach investigating & driving these down to the absolute minimum?

Should we at least put a big disclaimer as a log above this section saying "There should ideally be no items below this list. If you're considering adding more, please reconsider whether there's a real bug we should be addressing instead" or similar?

Reply:
> If I understand correctly, we will basically ignore these messages but they could still occur in real clusters and may represent real logical inconsistencies (possibly even bugs) in the agent code. 

My first impression is that most of these error logs should just be downgraded to warnings or lower. The exception so far are `globalDataSupport` and maybe `failedToListCRDs`. That would explain why they went unnoticed for so long (there's often no other impact).

> What do you think is the best way to approach investigating & driving these down to the absolute minimum?

Some of these issues should be easy to address (e.g., change log level) and are currently marked as `good-first-issues`. For some we may not be able to do anything about it (e.g., `opCannotBeFulfilled`). Given this is not high priority and we're all struggling to find cycles, it will probably take a bit of time before this is reduced to the absolute minimum.

> Should we at least put a big disclaimer as a log above this section saying "There should ideally be no items below this list. If you're considering adding more, please reconsider whether there's a real bug we should be addressing instead" or similar?

I was a bit hesitant to do that just now because some of these errors occur in very specific, racy conditions. That means we still see more errors that we need to add here in the week following the merge, despite my ~15 runs of the full CI in this pull request. For that time, I'd probably prefer if we accept new items in the list. We can always add the big warning next time we remove an item.